### PR TITLE
Updated documentation according the current project status

### PR DESCRIPTION
**Pull request by kobliha**

### CODE REVIEW

File: doc/architecture.txt [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
+2. Role based access management
 
-   Architecture backend uses D-Bus services that are actually accessing the system.
-   Those services have rights of system administrator (root).
+    The architecture uses Policykit to define rights to the configuration tasks
+    and check if current user has such right.
 
-   Only if user is allowed to perform certain task, D-Bus service is started and action is done. D-Bus is used only if non-root access system, so there is no speed penalty for old YaST usage (run as root).
+    Architecture backend uses D-Bus services that are actually accessing the
+    system. Those services have rights of system administrator (root).
```
Comment (jreidinger) [inline doc/architecture.txt:34]: Well, I think this is not clear enough and I consider splitting these part.

So when you are root, backend run as it is in same process.
And only for non-root dbus and policy kit is used. And I consider to refactor it out of base infrastructure. So every project can decide if they want role managent. So maybe we should specify here, that role based managent is optional and its implementation use DBus and policy-kit ( easily changable ).

Comment (jreidinger) [inline doc/architecture.txt:34]: So to summary I write it as 
Optional Role based access Management and describe what it means.

File: doc/architecture.txt [hunk 2]
```diff
@@ -49,58 +54,69 @@ Look at doc/architecture.svg diagram.
...
+                          the actual data to be written)
+    propose ({})        - propose new configuration (note: not yet decided if it
+                          should be in ylib or library above it, that just take
+                          input and return map for ylib modules)
 
 See https://github.com/yast/yast--/wiki/YLib-Use-Cases for more about YLib API.
 
-The Config Agent is working with the system, by which we mean it either modifies files or runs the scripts.
-There are 2 kinds of Config Agents, File Agents handle files, Script Agents are running the scripts.
+Config agent is works with system, by which we mean it either modifies files or
```
Comment (jreidinger) [inline doc/architecture.txt]: 'is works' looks like typo

File: doc/architecture.txt [hunk 3]
```diff
@@ -49,58 +54,69 @@ Look at doc/architecture.svg diagram.
...
-    YLib can be directly used from current YaST modules using yast2-ruby-bindings.
-    See 
-    	yast++lib-kerberos-client/examples/kerberos-example.ycp for simple example of calling YLib
-
-	yast--/yast/kerberos-client for whole YaST module where system access is replaced with YLib calls
+        yast++lib-kerberos-client/examples/kerberos_conf for an example
 
-    In mid-term, we want to replace Read/Write calls of YaST modules by YLib calls.
+    YLib can be directly used from current classical YaST modules using
+    yast2-ruby-bindings, see
```
Comment (jreidinger) [inline doc/architecture.txt]: thats not exact. We need simple layer between in ylib-ycp ( simple symlink ).

**Comment by jreidinger** [comment]
Hi, thanks for commit. Please look at my comments and comment it. Also please merge master into your branch so we can easy merge it ( usually `git pull upstream master` is good enough, if you follow githib tutorial ).

File: doc/architecture.txt (hunk 1 above)
Comment (kobliha) [inline doc/architecture.txt:34]: I'll write "optional", you may rephrase it later.

File: doc/architecture.txt (hunk 2 above)
Comment (kobliha) [inline doc/architecture.txt]: Definitely

File: doc/architecture.txt (hunk 3 above)
Comment (kobliha) [inline doc/architecture.txt]: Please be more specific or change yourself later.

**Pull request merged**